### PR TITLE
Add feature description placeholders to CI Visibility feature table

### PR DESCRIPTION
**Pull request by nikita-tkachenko-datadog**
### What does this PR do? What is the motivation?

* Add `ci-detail` HTML reusable component.
* Add brief descriptions for CI Visibility features.

We want to encourage users to migrate from JUnit XML uploads to the native instrumentation. The feature table's objective was this, but we thought that it lacked explanations on each of the features. This PR fixes it. 

See preview in https://docs-staging.datadoghq.com/nikita-tkachenko/ci-vis-feature-descriptions/continuous_integration/tests/#supported-features

### Merge instructions
- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/continuous_integration/tests/_index.md [hunk 1]
```diff
@@ -35,23 +35,23 @@ cascade:
...
+|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:----:|:----:|:------------------------:|:------:|:----:|:-----:|:------------------------:|
+| {{< ci-details title="Accurate time/durations results" >}}Microseconds resolution in time and duration results. This is usually not the case for JUnit XML reports.{{< /ci-details >}}         |  ✓   |  ✓   |            ✓             |   ✓    |  ✓   |   ✓   |            ✗             |
+| {{< ci-details title="Structured stack traces" >}} {{< /ci-details >}}                 |  ✓   |  ✓   |            ✓             |   ✓    |  ✓   |   ✓   |            ✗             |
+| {{< ci-details title="Distributed traces on integration tests" >}}Tests that make calls to external services instrumented with Datadog will show the full distributed trace in its test details.{{< /ci-details >}} |  ✓   |  ✓   |            ✓             |   ✗    |  ✓   |   ✓   |            ✗             |
```
Comment (nikita-tkachenko-datadog) [inline content/en/continuous_integration/tests/_index.md:42]:
```suggestion
| {{< ci-details title="Distributed traces on integration tests" >}}Tests that make calls to external services instrumented with Datadog will show the full distributed trace in their test details.{{< /ci-details >}} |  ✓   |  ✓   |            ✓             |   ✗    |  ✓   |   ✓   |            ✗             |
```

File: content/en/continuous_integration/tests/_index.md [hunk 2]
```diff
@@ -35,23 +35,23 @@ cascade:
...
+| {{< ci-details title="Agent-based reporting" >}}Ability to report test information via the Datadog Agent.{{< /ci-details >}}                   |  ✓   |  ✓   |            ✓             |   ✓    |  ✓   |   ✓   |            ✗             |
+| {{< ci-details title="Agentless reporting" >}}Ability to report test information without the need of the Datadog Agent.{{< /ci-details >}}                     |  ✓   |  ✓   |            ✓             |   ✓    |  ✗   |   ✓   |            ✓             |
+| {{< ci-details title="Test suite level visibility" >}}Visibility over the whole testing process, including session, module, suites and tests.{{< /ci-details >}}             |  ✓   |  ✓   |            ✓             |   ✓    |  ✗   |   ✓   |            ✓             |
```
Comment (nikita-tkachenko-datadog) [inline content/en/continuous_integration/tests/_index.md:45]: I think we can add a link to `#test-suite-level-visibility` section here

File: content/en/continuous_integration/tests/_index.md [hunk 3]
```diff
@@ -35,23 +35,23 @@ cascade:
...
+| {{< ci-details title="Manual API" >}}Ability to create CI Visibility events for test frameworks that are not currently supported by Datadog's automatic instrumentation.{{< /ci-details >}}                              |  ✓   |  ✓   |            ✓             |   ✗    |  ✗   |   ✓   |            ✗             |
```
Comment (nikita-tkachenko-datadog) [inline content/en/continuous_integration/tests/_index.md:46]:
```suggestion
| {{< ci-details title="Manual API" >}}Ability to programmatically create CI Visibility events for test frameworks that are not currently supported by Datadog's automatic instrumentation.{{< /ci-details >}}                              |  ✓   |  ✓   |            ✓             |   ✗    |  ✗   |   ✓   |            ✗             |
```

File: content/en/continuous_integration/tests/_index.md [hunk 4]
```diff
@@ -35,23 +35,23 @@ cascade:
...
+| {{< ci-details title="Codeowner by test" >}}Automatic detection of the owner of a test based on the CODEOWNERS file.{{< /ci-details >}}                       |  ✓   |  ✓   |            ✓             |   ✗    |  ✗   |   ✓   |            ✗             |
+| {{< ci-details title="Source code start/end" >}}Automatic report of the start and end line of a test.{{< /ci-details >}}         |  ✓   |  ✓   | ✓&nbsp;(only&nbsp;start) |   ✗    |  ✗   |   ✓   | ✓&nbsp;(only&nbsp;start) |
+| {{< ci-details title="CI and git info" >}}Automatic collection of git and CI environment metadata, such as CI provider, git commit SHA or pipeline URL.{{< /ci-details >}}                         |  ✓   |  ✓   |            ✓             |   ✓    |  ✓   |   ✓   |            ✓             |
+| {{< ci-details title="Git metadata upload" >}}Automatic upload of git tree information used for Intelligent Test Runner.{{< /ci-details >}}                     |  ✓   |  ✓   |            ✓             |   ✓    |  ✗   |   ✓   |            ✓             |
```
Comment (nikita-tkachenko-datadog) [inline content/en/continuous_integration/tests/_index.md:50]: I think we can add a link to Intelligent Test Runner docs page here (and in the next row too).

File: content/en/continuous_integration/tests/_index.md [hunk 5]
```diff
@@ -35,23 +35,23 @@ cascade:
...
+| {{< ci-details title="Test suite level visibility" >}}Visibility over the whole testing process, including session, module, suites and tests.{{< /ci-details >}}             |  ✓   |  ✓   |            ✓             |   ✓    |  ✗   |   ✓   |            ✓             |
+| {{< ci-details title="Manual API" >}}Ability to create CI Visibility events for test frameworks that are not currently supported by Datadog's automatic instrumentation.{{< /ci-details >}}                              |  ✓   |  ✓   |            ✓             |   ✗    |  ✗   |   ✓   |            ✗             |
+| {{< ci-details title="Codeowner by test" >}}Automatic detection of the owner of a test based on the CODEOWNERS file.{{< /ci-details >}}                       |  ✓   |  ✓   |            ✓             |   ✗    |  ✗   |   ✓   |            ✗             |
+| {{< ci-details title="Source code start/end" >}}Automatic report of the start and end line of a test.{{< /ci-details >}}         |  ✓   |  ✓   | ✓&nbsp;(only&nbsp;start) |   ✗    |  ✗   |   ✓   | ✓&nbsp;(only&nbsp;start) |
```
Comment (nikita-tkachenko-datadog) [inline content/en/continuous_integration/tests/_index.md:48]:
```suggestion
| {{< ci-details title="Source code start/end" >}}Automatic report of the start and end lines of a test.{{< /ci-details >}}         |  ✓   |  ✓   | ✓&nbsp;(only&nbsp;start) |   ✗    |  ✗   |   ✓   | ✓&nbsp;(only&nbsp;start) |
```

File: content/en/continuous_integration/tests/_index.md [hunk 6]
```diff
@@ -35,23 +35,22 @@ cascade:
...
+| {{< ci-details title="Git metadata upload" >}}Automatic upload of git tree information used for <a href="/continuous_integration/intelligent_test_runner">Intelligent Test Runner</a>.{{< /ci-details >}}                     |  ✓   |  ✓   |            ✓             |   ✓    |  ✗   |   ✓   |            ✓             |
+| {{< ci-details title="Intelligent Test Runner" >}}Capability to enable <a href="/continuous_integration/intelligent_test_runner">Intelligent Test Runner</a>, which intelligently skips tests based on code coverage and git metadata.{{< /ci-details >}}                 |  ✓   |  ✓   |            ✓             |   ✓    |  ✗   |   ✓   |            ✗             |
+| {{< ci-details title="Code coverage support" >}}Ability to report total code coverage metrics.{{< /ci-details >}}                   |  ✓   |  ✓   |            ✓             |   ✗    |  ✗   |   ✓   |            ✓ (manual)             |
+| {{< ci-details title="Benchmark tests support" >}}Automatic detection performance statistics for benchmark tests.{{< /ci-details >}}                 |  ✓   |  ✗   |            ✗             |   ✓    |  ✗   |   ✓   |            ✗             |
```
Comment (nikita-tkachenko-datadog) [inline content/en/continuous_integration/tests/_index.md:52]:
```suggestion
| {{< ci-details title="Benchmark tests support" >}}Automatic detection of performance statistics for benchmark tests.{{< /ci-details >}}                 |  ✓   |  ✗   |            ✗             |   ✓    |  ✗   |   ✓   |            ✗             |
```

**Review by alai97 — APPROVED** [review]
What a clever improvement, this is really awesome, thanks!

**Pull request merged**